src/modm/driver/adc/ads7828.hpp

```diff
@@ -0,0 +1,171 @@
+// coding: utf-8
+// ----------------------------------------------------------------------------
+/*
+ * Copyright (c) 2022, Jónas Holm Wentzlau
+ * Copyright (c) 2022, Jonas Kazem Andersen
+ * Copyright (c) 2022, Rasmus Kleist Hørlyck Sørensen
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_ADS7828_HPP
+#define MODM_ADS7828_HPP
+
+#include <modm/architecture/interface/i2c_device.hpp>
+
+namespace modm
+{
+
+/// @ingroup modm_driver_ads7828
+struct ads7828
+{
+protected:
+    enum class
+    CommandByte : uint8_t
+    {
+        SD = Bit7,
+        C2 = Bit6,
+        C1 = Bit5,
+        C0 = Bit4,
+        PD1 = Bit3,
+        PD0 = Bit2,
+    };
+    MODM_FLAGS8(CommandByte);
+
+public:
+    enum class
+    InputChannel : uint8_t
+    {
+        // Differential Inputs
+        Ch0Ch1 = 0u,
+        Ch2Ch3 = int(CommandByte::C0),
+        Ch4Ch5 = int(CommandByte::C1),
+        Ch6Ch7 = int(CommandByte::C1) | int(CommandByte::C0),
+        Ch1Ch0 = int(CommandByte::C2),
+        Ch3Ch2 = int(CommandByte::C2) | int(CommandByte::C0),
+        Ch5Ch4 = int(CommandByte::C2) | int(CommandByte::C1),
+        Ch7Ch6 = int(CommandByte::C2) | int(CommandByte::C1) | int(CommandByte::C0),
+
+        // Single-Ended Inputs
+        Ch0 = int(CommandByte::SD),
+        Ch1 = int(CommandByte::SD) | int(CommandByte::C2),
+        Ch2 = int(CommandByte::SD) | int(CommandByte::C0),
+        Ch3 = int(CommandByte::SD) | int(CommandByte::C2) | int(CommandByte::C0),
+        Ch4 = int(CommandByte::SD) | int(CommandByte::C1),
+        Ch5 = int(CommandByte::SD) | int(CommandByte::C2) | int(CommandByte::C1),
+        Ch6 = int(CommandByte::SD) | int(CommandByte::C1) | int(CommandByte::C0),
+        Ch7 = int(CommandByte::SD) | int(CommandByte::C2) | int(CommandByte::C1) | int(CommandByte::C0),
+    };
+    typedef Configuration<CommandByte_t, InputChannel, Bit7 | Bit6 | Bit5> InputChannel_t;
```

Comment:
```suggestion
    using InputChannel_t = Configuration<CommandByte_t, InputChannel, Bit7 | Bit6 | Bit5>;
```

In modern C++ `using` should be preferred to `typedef` for aliases, see [C++ core guidelines T.43](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rt-using) for reasoning.

@salkinium We should set up `clang-tidy` in the CI to enforce such things.